### ISSUE TITLE
Add support for modifying PR draft status

### ISSUE THREAD

**Issue by charlax** [PyGithub/PyGithub#1461]
`PullRequest.edit` does not allow a `draft` attribute:

```python
def edit(
        self,
        title=github.GithubObject.NotSet,
        body=github.GithubObject.NotSet,
        state=github.GithubObject.NotSet,
        base=github.GithubObject.NotSet,
    ):
```

**Comment by FlorentClarret** [comment]
I might be wrong, but it looks like this is not yet possible to update the draft attribute according to the github documentation According to the github documentation, this is not yet available on their side : https://developer.github.com/v3/pulls/#update-a-pull-request

**Comment by charlax** [comment]
I think you're right, I tried making a plain HTTP request and it didn't work. Also, in the documentation they hint at this being available only v4. 

Bummer.

**Comment by sfdye** [comment]
![image](https://user-images.githubusercontent.com/1016390/79286684-eefe0900-7ef3-11ea-8a75-948022fc54b2.png)
I can see it 😳

**Comment by FlorentClarret** [comment]
It's definitely in the output and the draft attribute is already in the PullRequest class, however it does not seem possible to update it through the PATCH method (I also tried when @charlax submitted the issue)

**Comment by sfdye** [comment]
Maybe need some custom headers?

**Comment by s-t-e-v-e-n-k** [comment]
Nope, custom headers won't help, the docs don't list it at all for the PATCH call. :-(

**Comment by nitishagar** [comment]
@sfdye I confirmed what @s-t-e-v-e-n-k has reported here from the docs: https://docs.github.com/en/rest/reference/pulls#update-a-pull-request

In this case do we close this issue or how we approached such request in the past?

**Comment by danieljamesscott** [comment]
Looks like this may be supported now? https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request

**Comment by VardaQuraishi** [comment]
Is this issue still available to work on? @charlax

**Comment by EnricoMi** [comment]
@VardaQuraishi yes, please go for it!

**Comment by mig5** [comment]
> Looks like this may be supported now? https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request

Looks to me like the 'state' param is still just 'open' or 'closed' ?

**Comment by cperrin88** [comment]
As far as I can see there is only a GraphQL mutation to [change the state of the PR](https://docs.github.com/en/graphql/reference/mutations#markpullrequestreadyforreview).

**Comment by EnricoMi** [comment]
PyGithub now has its first GraphQL call, see https://github.com/PyGithub/PyGithub/commit/232df79a2947dc96363bdc6f51a6f446ff2b241f#diff-90017b18548bd848900b7cffd7fe86a610dc20e5d228663dfb429a47cdf65a20R767-R772 for reference.

**Comment by tkrop** [comment]
This sound like a very quick and compatible extension possible. A quick win that would make integration and tooling around pull request much simpler and better.
